brokers/nats/publish.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+// Package nats contains NATS message publisher implementation.
+package nats
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/mainflux/mainflux"
+	"github.com/mainflux/mainflux/brokers"
+	"github.com/mainflux/mainflux/logger"
+	"github.com/nats-io/nats.go"
+)
+
+const prefix = "channel"
+
+var _ brokers.MessagePublisher = (*natsPub)(nil)
+
+type natsPub struct {
+	conn *nats.Conn
+}
+
+// NewPublisher NATS message publisher.
+func NewPublisher(url string, log logger.Logger) brokers.MessagePublisher {
```

Comment:
It would be better to return an error because `os.Exit(1)` should be reserved for the `main` method. That way, we also don't need logger param, but we'll log error in caller method (i.e. `main`).